diff -Naur chkconfig-1.12.tpg/Makefile chkconfig-1.12.tpg.tpg/Makefile
--- chkconfig-1.12.tpg/Makefile	2020-04-08 10:34:01.462051637 +0000
+++ chkconfig-1.12.tpg.tpg/Makefile	2020-04-08 12:47:05.683098701 +0000
@@ -10,6 +10,7 @@
 MANDIR = /usr/man
 ALTDIR = /var/lib/alternatives
 ALTDATADIR = /etc/alternatives
+SYSTEMDUTILDIR = $(shell pkg-config --variable=systemdutildir systemd)
 SUBDIRS = po
 
 OBJS=chkconfig.o leveldb.o
@@ -55,10 +56,10 @@
 	[ -d $(DESTDIR)/$(MANDIR)/man8 ] || mkdir -p $(DESTDIR)/$(MANDIR)/man8
 	[ -d $(DESTDIR)/$(ALTDIR) ] || mkdir -p -m 755 $(DESTDIR)/$(ALTDIR)
 	[ -d $(DESTDIR)/$(ALTDATADIR) ] || mkdir -p -m 755 $(DESTDIR)/$(ALTDATADIR)
-	[ -d $(DESTDIR)/usr/lib/systemd ] || mkdir -p -m 755 $(DESTDIR)/usr/lib/systemd
+	[ -d $(DESTDIR)/$(SYSTEMDUTILDIR) ] || mkdir -p -m 755 $(DESTDIR)/$(SYSTEMDUTILDIR)
 
 	install -m 755 $(PROG) $(DESTDIR)/$(BINDIR)/$(PROG)
-	ln -s ../../../$(BINDIR)/$(PROG) $(DESTDIR)/usr/lib/systemd/systemd-sysv-install
+	ln -s ../../../$(BINDIR)/$(PROG) $(DESTDIR)/$(SYSTEMDUTILDIR)/systemd-sysv-install
 
 	install -m 755 ntsysv $(DESTDIR)/$(SBINDIR)/ntsysv
 	install -m 755 alternatives $(DESTDIR)/$(SBINDIR)/alternatives
